central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen.go

```diff
@@ -0,0 +1,32 @@
+package complianceReportgenerator
+
+import (
+	"bytes"
+	"context"
+
+	checkResults "github.com/stackrox/rox/central/complianceoperator/v2/checkresults/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/notifier"
+)
+
+// ReportGenerator interface is used to generate compliance report and send email notification.
+//
+//go:generate mockgen-wrapper
+type ComplianceReportGenerator interface {
+	// ProcessReportRequest will generate a csv report and send notification via email to attached scan config notifiers.
+	ProcessReportRequest(req *ComplianceReportRequest) error
+
+	getDataForReport(req *ComplianceReportRequest) (*resultEmail, error)
```

Comment:
It seems odd that we have private funcs in an interface. Should this just be a struct and not an interface?

Reply:
I have removed private functions